### PR TITLE
rework dependency type to `provides`

### PR DESCRIPTION
**Pull request by jkowalleck**
changes as discussed here: https://github.com/CycloneDX/specification/pull/347#discussion_r1466522226

### CODE REVIEW

File: schema/bom-1.6.proto [hunk 1]
```diff
@@ -171,16 +171,12 @@ enum DataFlowDirection {
...
-    DEPENDENCY_TYPE_USES = 1;
-  }
   // References a component or service by the its bom-ref attribute
   string ref = 1;
   // The bom-ref identifiers of the components or services that are dependencies of this dependency object.
   repeated Dependency dependencies = 2;
-  // Defines and characterizes the type of dependency
-  optional DependencyType type = 3;
+  // The bom-ref identifiers of the components or services that define a given specification or standard, which are provided or implemented by this dependency object.
+  repeated string provides = 3;
```
Comment (prabhu) [inline schema/bom-1.6.proto:179]: `provides` is a verb. Can we use a bit generic like `capabilities` to indicate a state and not an action?

Comment (jkowalleck) [reply schema/bom-1.6.proto:179]: we used "depends on" on the sibling ...

Comment (jkowalleck) [reply schema/bom-1.6.proto:179]: > `provides` is a verb. Can we use a bit generic like `capabilities` to indicate a state and not an action?

for protobuff, I'd agree. a noun would go with the current naming conventions.

for JSON/XML, the current naming conventions are verbs.

Comment (stevespringett) [reply schema/bom-1.6.proto:179]: Sounds like your suggestion for `capabilities` @prabhu would be a good fit for what we're doing with CDX 1.7 - specifically architecture and blueprints.

**Review by stevespringett — APPROVED** [review]

**Pull request merged**